tritium-annotations/src/main/java/com/palantir/tritium/annotations/PerformanceSensitive.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Elements annotated with {@link PerformanceSensitive} indicate that there are performance considerations to balance
+ * when modifying this element.
+ */
+@Target({ElementType.TYPE, ElementType.METHOD, ElementType.MODULE})
+@Retention(RetentionPolicy.RUNTIME)
```

Comment:
Do we need runtime retention? If we're using this as a target to opt into specific static analysis rules, SOURCE retention should be sufficient (and doesn't require us to include jars at runtime)

Reply:
I was thinking we could do bytecode inspection, but probably not necessary now.

Reply:
We could still write error-prone checks validating certain behaviors (e.g. restricting use of streams within certain scopes).

Reply:
Source retention should be fine for this case, as long as we’re inspecting the annotated element. I imagine calling into a perf sensitive method shouldn’t be relevant to the caller, so bytecode retention shouldn’t buy us much unless we want to add specific instrumentation 👍